[FSDP] Ignore buffers that are non-persistent.

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85761
* __->__ #85740

A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

fegin committed:
[FSDP] Ignore the buffers that are non-persistent.

A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Ignore buffers that are non-persistent."


A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]

awgu reviewed (approved):
LGTM! (TIL about `persistent=False`.)

fegin committed:
Update on "[FSDP] Ignore buffers that are non-persistent."


A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Ignore buffers that are non-persistent."


A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Ignore buffers that are non-persistent."


A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Ignore buffers that are non-persistent."


A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Ignore buffers that are non-persistent."


A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Ignore buffers that are non-persistent."


A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Ignore buffers that are non-persistent."


A buffer can be registered as non-persistent. A non-persistent buffer won't be in the state_dict.

Differential Revision: [D39858689](https://our.internmc.facebook.com/intern/diff/D39858689/)

[ghstack-poisoned]
